core/tests/protractor_desktop/topicAndStoryViewer.js

```diff
@@ -0,0 +1,159 @@
+      'Topic SPF1', 'topic-spf-one', 'Description', false);
+    await topicEditorPage.submitTopicThumbnail(Constants.TEST_SVG_PATH, true);
+    await topicEditorPage.updateMetaTagContent('topic meta tag');
+    await topicEditorPage.saveTopic('Added thumbnail.');
+    var url = await browser.getCurrentUrl();
+    var topicId = url.split('/')[4].slice(0, -1);
+    await general.closeCurrentTabAndSwitchTo(handle);
+    await adminPage.editConfigProperty(
+      'The details for each classroom page.',
+      'List',
+      async function(elem) {
+        elem = await elem.editItem(0, 'Dictionary');
+        elem = await elem.editEntry(4, 'List');
+        elem = await elem.addItem('Unicode');
+        await elem.setValue(topicId);
+      });
+
+    await topicsAndSkillsDashboardPage.get();
+    (
+      await
+      topicsAndSkillsDashboardPage.createSkillWithDescriptionAndExplanation(
+        'Skill SPF1', 'Concept card explanation', false));
+    await topicsAndSkillsDashboardPage.get();
+    await topicsAndSkillsDashboardPage.navigateToSkillsTab();
+    await topicsAndSkillsDashboardPage.assignSkillToTopic(
+      'Skill SPF1', 'Topic SPF1');
+    await topicsAndSkillsDashboardPage.get();
+    await topicsAndSkillsDashboardPage.editTopic('Topic SPF1');
+    await topicEditorPage.addSubtopic(
+      'Subtopic SPF1', 'subtopic-spf-one', Constants.TEST_SVG_PATH,
+      'Subtopic content');
+    await topicEditorPage.saveTopic('Added subtopic.');
+
+    await topicEditorPage.navigateToTopicEditorTab();
+    await topicEditorPage.navigateToReassignModal();
+
+    await topicEditorPage.dragSkillToSubtopic('Skill SPF1', 0);
+    await topicEditorPage.saveRearrangedSkills();
+    await topicEditorPage.saveTopic('Added skill to subtopic.');
+    await topicEditorPage.publishTopic();
+    await topicsAndSkillsDashboardPage.editTopic('Topic SPF1');
+    await topicEditorPage.createStory(
+      'Story SPF1', 'story-player-spf-one', 'Story description',
+      Constants.TEST_SVG_PATH);
+    await storyEditorPage.updateMetaTagContent('story meta tag');
+    for (var i = 0; i < 3; i++) {
+      await storyEditorPage.createNewChapter(
+        `Chapter ${i}`, dummyExplorationIds[i], Constants.TEST_SVG_PATH);
+      await storyEditorPage.navigateToChapterWithName(`Chapter ${i}`);
+      await storyEditorPage.changeNodeDescription('Chapter description');
+      await storyEditorPage.changeNodeOutline(
+        await forms.toRichText(`outline ${i}`));
+      await storyEditorPage.navigateToStoryEditorTab();
+    }
+    await storyEditorPage.saveStory('First save');
+    await storyEditorPage.publishStory();
+    await users.logout();
+    await topicAndStoryViewerPage.get(
+      'math', 'topic-spf-one', 'story-player-spf-one');
+    await topicAndStoryViewerPage.expectLockedChaptersToBe(2);
+    await topicAndStoryViewerPage.goToChapterIndex(0);
+    await explorationPlayerPage.submitAnswer('Continue', null);
```

Comment:
Should you check below this that the log in and sign up bettons are visible?

Reply:
Oh, I see you do it in the `completeLoginFlowFromStoryViewerPage` it would probably make sense to check it here and then handle the login in the users.